solutions/deploy/cloud-pak-deployer/scripts/uninstall.sh

```diff
@@ -1,8 +1,62 @@
 #!/bin/bash
 
-${OC} create -f "${JOB_UNINSTALL_CPD_FILENAME}"
+# Create manifest for uninstall
+UNINSTALL_MANIFEST="/tmp/uninstall.yml"
+cat > "${UNINSTALL_MANIFEST}" << EOF
+apiVersion: batch/v1
+kind: Job
+metadata:
+  labels:
+    app: ${CLOUD_PAK_DEPLOYER_JOB_LABEL}-uninstall-cpd
+  name: ${CLOUD_PAK_DEPLOYER_UNINSTALL_JOB_NAME}
+  namespace: ${CLOUD_PAK_DEPLOYER_NAMESPACE_NAME}
+spec:
+  parallelism: 1
+  completions: 1
+  backoffLimit: 0
+  template:
+    metadata:
+      name: ${CLOUD_PAK_DEPLOYER_UNINSTALL_JOB_NAME}
+      labels:
+        app: ${CLOUD_PAK_DEPLOYER_JOB_LABEL}-uninstall-cpd
+    spec:
+      imagePullSecrets:
+        ${CLOUD_PAK_DEPLOYER_IMAGE_SECRET}
+      containers:
+        - name: ${CLOUD_PAK_DEPLOYER_UNINSTALL_JOB_NAME}
+          image: ${CLOUD_PAK_DEPLOYER_IMAGE}
+          imagePullPolicy: Always
+          terminationMessagePath: /dev/termination-log
+          terminationMessagePolicy: File
+          env:
+            - name: CONFIG_DIR
+              value: /Data/cpd-config
+            - name: STATUS_DIR
+              value: /Data/cpd-status
+          volumeMounts:
+            - name: config-volume
+              mountPath: /Data/cpd-config/config
+            - name: status-volume
+              mountPath: /Data/cpd-status
+          command: ["/bin/sh", "-xc"]
+          args:
+            - /cloud-pak-deployer/scripts/cp4d/cp4d-delete-instance.sh cpd <<< "y"
+      restartPolicy: Never
+      securityContext:
+        runAsUser: 0
+      serviceAccountName: ${CLOUD_PAK_DEPLOYER_SERVICE_ACCOUNT_NAME}
+      volumes:
+        - name: config-volume
+          configMap:
+            name: ${CLOUD_PAK_DEPLOYER_CONFIG_MAP_NAME}
+        - name: status-volume
+          persistentVolumeClaim:
+            claimName: ${CLOUD_PAK_DEPLOYER_PERSISTENT_VOLUME_CLAIM_NAME}
+EOF
+
+${OC} create -f "${UNINSTALL_MANIFEST}"
```

Comment:
could we use a native terraform way here - eg: kube provider? 

Also the yaml here won't get linted as embedded into bash

Reply:
Problem is the lifecycle of this execution.  Only is done during destroy.  Would be nice if there is a native route through the kube provider but haven't seen a way

Reply:
I see - I think this could be done through a helm pre-delete hook. the helm provider will install the resource but not execute the job. the job is only executed when the helm release is removed on terraform destroy.